### PR TITLE
#168049997 An Email should be sent to a user upon successful registration. User should be able to receive a verification link via email upon successful signup

### PR DESCRIPTION
**Pull request by amilykassim**
#### What does this PR do?
- An email should be sent to a user upon successful registration.
#### Description of Task to be completed?
- Create a SendGrid account and create an API KEY
- Send a verification email to a user once he/she finishes to signup
- Activate an account as verified once a user confirms his/her email
#### How should this be manually tested?
1. Run `git checkout ft-email-verification-168049997`  to your terminal.

2. Run `npm install`

3. Remember to add the variable `SENDGRID_API_KEY` to your `.env` file with the corresponding value.

4. Run `npm run migrate:refresh` to create users tables and models

5. Run `npm run dev`

6. Sign up a user using this endpoint `/api/v1/auth/register` using `POST` method, you will also need to add in an `email`, `password`, `firstname`, `lastname` , `username` fields

7. After sign up, you'll receive a confirmation email and then click on the `confirm email` button

8. You will receive a response after it is complete

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168049997](https://www.pivotaltracker.com/story/show/168049997)
#### What are the relevant Github Issues?
N/A
#### Screenshots (if appropriate)
*Register a user*
Screen Shot 2019-09-12 at 10 43 27

*Email verification*
Screen Shot 2019-09-12 at 10 42 49

N/A
#### Questions:
N/A

### CODE REVIEW

File: src/services/UserServices.js [hunk 1]
```diff
@@ -25,6 +30,23 @@ class UserService {
     if (!user) return null;
     return user.dataValues;
   }
+
```
Comment (gik-hub) [inline src/services/UserServices.js]: Awesome @amilykassim.

It seems like the sendConfirmationEmail method is currently working with your email only. Please make it a variable containing the specific user who has registered and needs to confirm their password.

**Review by gik-hub — CHANGES_REQUESTED** [review]

File: src/services/UserServices.js (hunk 1 above)
Comment (amilykassim) [reply src/services/UserServices.js]: Great!

**Review by gik-hub — APPROVED** [review]
Great, work man  :fireworks:

**Review by Jordybastien — APPROVED** [review]
LGTM

File: src/services/UserServices.js [hunk 2]
```diff
@@ -63,6 +68,28 @@ class UserService {
...
+   * @param  {String} token
+   * @param  {String} email
+   * @returns {Object} result;
+   */
+  static async sendConfirmationEmail(token, email) {
+    sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+    let message = {
+      to: email,
+      from: 'barefoot@gmail.com',
```
Comment (Paulvitalis200) [inline src/services/UserServices.js]: Put the `from` field inside an environment variable.

Comment (amilykassim) [reply src/services/UserServices.js]: Great, let me work on that!!

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**